XenAdmin/Wizards/CrossPoolMigrateWizard/CrossPoolMigrateWizard.cs

```diff
@@ -239,8 +244,30 @@ protected override void FinishWizard()
 
                 if (wizardMode == WizardMode.Move && IsIntraPoolMove(pair))
                     new VMMoveAction(vm, pair.Value.Storage, target).RunAsync();
-                else 
-                    new VMCrossPoolMigrateAction(vm, target, SelectedTransferNetwork, pair.Value, wizardMode == WizardMode.Copy).RunAsync();
+                else
+                {
+                    var isCopy = wizardMode == WizardMode.Copy;
+
+                    if (_resumeAfterMigrate)
+                    {
+                        var title = VMCrossPoolMigrateAction.GetTitle(vm, target, isCopy);
+                        var startDescription = isCopy ? Messages.ACTION_VM_COPYING: Messages.ACTION_VM_MIGRATING;
+                        var endDescription = isCopy ? Messages.ACTION_VM_COPIED: Messages.ACTION_VM_MIGRATED;
+
+                        var actions = new List<AsyncAction>()
+                        {
+                            new VMCrossPoolMigrateAction(vm, target, SelectedTransferNetwork, pair.Value, isCopy),
+                            new ResumeAndStartVMsAction(vm.Connection, target, new List<VM>{vm}, new List<VM>(), null, null)
+                        };
+
+                        new MultipleAction(vm.Connection, title, startDescription, endDescription,
+                            actions, false, false, true).RunAsync();

```

Comment:
We should probably use the MultipleAction with suppressHistory parameter set to true, so that the multiple action does not appear on the Events page, only the subactions. In this case the title, startDescription and stopDescription are not that important, but it is still good to have them properly set.